ggshield/core/scan/commit.py

```diff
@@ -73,6 +75,57 @@ def parser(commit: "Commit") -> Iterable[Scannable]:
 
         return Commit(sha, parser, info)
 
+    @staticmethod
+    def from_merge(
+        exclusion_regexes: Optional[Set[Pattern[str]]] = None,
+        merge_branch: str = "MERGE_HEAD",
+        cwd: Optional[Path] = None,
+    ) -> "Commit":
+
+        def get_file_sha_from_ls_tree(
+            ref: str,
+            list_diff_files: List[str] = [],
+        ) -> Iterable[Tuple[str, str]]:
+            ls_tree = git(["ls-tree", "-z", ref] + list_diff_files, cwd=cwd)
+            for line in ls_tree.split("\x00")[:-1]:
+                mode, type, sha, path = line.split()
+                yield (path, sha)
+
+        # Get files modified and staged
+        diff_output = git(["diff", "--name-only", "--staged", "-z"], cwd=cwd)
+        list_diff_files = diff_output.split("\x00")[:-1]  # remove the trailing \x00
+
+        dict_path_to_shas = defaultdict(set)
```

Comment:
Same as `list_diff_files` here: `path_to_shas` would be nicer, but actually `shas_for_path` would be even nicer, because here `_to_` is used to express a mapping, but further down (`files_to_scan`) `_to_` is used because it's before a verb (I was stuck a bit wondering how a mapping could be a set 😅).

Reply:
Sorry and thanks for the pointers on how to improve naming!